Api/Services/IngredientsService.cs

```diff
@@ -30,22 +31,56 @@ public async Task<Result<IngredientVM>> CreateIngredientAsync(CreateIngredientRe
         {
             return result;
         }
+        var menuItem = await dbContext.MenuItems.Include(m => m.Ingredients).Where(m => m.Id == request.MenuItem.MenuItemId).FirstOrDefaultAsync();
+        if (menuItem is null)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = nameof(request.MenuItem),
+                ErrorMessage = ErrorCodes.NotFound,
+                ErrorCode = ErrorCodes.NotFound
+            };
+        }
+
+        var restaurant = await dbContext.Restaurants.Include(r => r.MenuItems).Include(r => r.Group).Where(r => r.Group.OwnerId == userId && r.MenuItems.Contains(menuItem)).FirstOrDefaultAsync();
```

Comment:
nie trzeba pisać include dla rzeczy których używasz w where

generalnie nie potrzebujesz całej restauracji, tylko sprawdzasz, czy istnieje, to możesz użyć AnyAsync